src/app/data-model-provider/MetadataTypes.h

```diff
@@ -89,11 +89,41 @@ struct AttributeEntry
     AttributeId attributeId;
     BitFlags<AttributeQualityFlags> flags;
 
+
+    // Constructor
+    AttributeEntry() : readPrivilege(0), writePrivilege(0) {}
+
+    void SetReadPrivilege(Access::Privilege r)
+    {
+        readPrivilege = chip::to_underlying(r);
+    }
+
+    Access::Privilege GetReadPrivilege() const
+    {
+        return static_cast< Access::Privilege >(readPrivilege);
+    }
+
+    void SetWritePrivilege(Access::Privilege w)
+    {
+        writePrivilege = chip::to_underlying(w);
+    }
+
+    Access::Privilege GetWritePrivilege() const
+    {
+        return static_cast< Access::Privilege >(writePrivilege);
+    }
+
+    bool readPrivilegeHasValue() {return readPrivilege;}
+    bool writePrivilegeHasValue() {return writePrivilege;}
+
+
+    private:
+
     // read/write access will be missing if read/write is NOT allowed
     //
     // NOTE: this should be compacted for size
-    std::optional<Access::Privilege> readPrivilege;  // generally defaults to View if readable
-    std::optional<Access::Privilege> writePrivilege; // generally defaults to Operate if writable
+    std::underlying_type_t<Access::Privilege> readPrivilege : 5 ;  // generally defaults to View if readable
```

Comment:
this needs to be a better combination. Maybe create some struct and please combine them with flags.
Assert that the final size is at most 4 bytes, so we show file savings.

Combination of  flags and r/w privileges has to go in the same PR.

Is there a way to assert that Access::Privilege fits within the bits that we assign to it?